dhis-2/dhis-api/src/main/java/org/hisp/dhis/icon/IconOperationParams.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2004-2023, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.icon;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Represents query parameters sent to {@link org.hisp.dhis.icon.IconService}.
+ *
+ * @author Zubair Asghar
+ */
+@Data
+@NoArgsConstructor
+public class IconOperationParams {
```

Comment:
I have just realized it is not possible to filter a list of "key", is it? 
In Android, the workflow would be:
1. Download user-related metadata (programs assigned, data elements, etc)
2. Collect the list of icons assigned to that metadata and filter those whose icon has not been downloaded.
3. Request the "/api/icons" to download the information about those icons
4. Download the file resources

In the step 3, we could do a single request if it were possible to filter by key (something similar to `filter=id:in:id1,id2,id3,..` in metadata). I assume we need the JsonIcon object in order to be able to download the actual file resource. Or is it possible to download the file resource without downloading the JsonIcon object?

Reply:
To download the file you need to hit this endpoint` /{key}/icon`. `{key}` is the fileResourceUid that you get after downloading CustomIcon.

Reply:
In that case, I would suggest to add a filter such as `key=icon1,icon2,icon3,...` in order to download a list of CustomIcons in a single query instead of doing a request per icon. It doesn't necessarily have to be part of this PR

Reply:
yes this is possible using  `api/icons?keys=icon1,icon2,icon3`